buster/documents/deeplake.py

```diff
@@ -0,0 +1,93 @@
+import logging
+from typing import Optional
+
+import openai
+import pandas as pd
+from deeplake.core.vectorstore import VectorStore
+from utils import zip_contents
+
+logger = logging.getLogger(__name__)
+logging.basicConfig(level=logging.INFO)
+
+
+def openai_embedding_function(texts, model="text-embedding-ada-002"):
+    if isinstance(texts, str):
+        texts = [texts]
+
+    texts = [t.replace("\n", " ") for t in texts]
+    return [data["embedding"] for data in openai.Embedding.create(input=texts, model=model)["data"]]
+
+
+def extract_metadata(df: pd.DataFrame) -> dict:
+    """extract the metadata from the dataframe in deeplake dict format"""
+    metadata = df.apply(
+        lambda x: {
+            "url": x.url,
+            "source": x.source,
+            "title": x.title,
```

Comment:
you should extract everything except the content I think. Otherwise it wouldn't work for the OECD country or year fields for example.

Reply:
good idea

Reply:
fixed in latest commits